src/hostnet_test/test_half_close.ml

```diff
@@ -166,6 +166,105 @@ let test_host_half_close () =
       )
   end
 
+let test_connect_invalid_port () =
+  Host.Main.run begin
+    Slirp_stack.with_stack ~pcap:"test_connect_invalid_port.pcap" (fun _ stack -> with_server (fun _ ->
+        Lwt.return_unit
+      ) (fun server ->
+            (* Now that a server is running, connect to an invalid port and ensure it fails quickly *)
+            let open Slirp_stack in
+            let ip = Ipaddr.V4.localhost in
+            let port = server.Server.port + 1 in
```

Comment:
Perhaps on the Windows machine `server.Server.port + 1` was used?
```
**** Starting test TCP: test server invalid port
...
9.09694 [ERROR] Connected to localhost:1986
```
Maybe quickly allocating and deallocating a port will be more reliable, like
```ocaml
with_server (fun _ -> Lwt.return_unit) (fun server -> Server.destroy server >>= fun () -> Lwt.return server.Server.port)
>>= fun invalid_port ->
```

Reply:
Probably, in that case, it becomes pretty similar than the next test which ensure connection succeeds first and then destroys the server